plugins/module_utils/network/ios/config/snmp_server/snmp_server.py

```diff
@@ -77,72 +77,121 @@ def __init__(self, module):
             "views",
         ]
         self.complex_parsers = [
+            "traps.aaa_server",
             "traps.auth_framework",
             "traps.bfd",
             "traps.bgp",
+            "traps.bgp.cbgp2",
             "traps.bridge",
+            "traps.bulkstat",
+            "traps.call_home",
             "traps.casa",
+            "traps.cef",
             "traps.cnpd",
             "traps.config",
             "traps.config_copy",
             "traps.config_ctid",
+            "traps.cpu",
             "traps.dhcp",
+            "traps.dlsw",
             "traps.eigrp",
-            "traps.entity",
             "traps.energywise",
+            "traps.entity",
+            "traps.entity_diag",
+            "traps.entity_perf",
+            "traps.entity_state",
+            "traps.envmon",
+            "traps.errdisable",
+            "traps.ether_oam",
+            "traps.ethernet.cfm.alarm",
+            "traps.ethernet.cfm.cc",
+            "traps.ethernet.cfm.crosscheck",
+            "traps.ethernet.evc",
             "traps.event_manager",
+            "traps.flash",
+            "traps.flex_links",
+            "traps.firewall",
             "traps.flowmon",
+            "traps.frame_relay",
+            "traps.frame_relay.subif",
             "traps.fru_ctrl",
             "traps.hsrp",
-            "traps.ipsla",
-            "traps.isis",
-            "traps.msdp",
-            "traps.mvpn",
-            "traps.mpls_vpn",
```

Comment:
I can see there are a few parsers and attributes that have be removed/updated to a different parser, we cannot directly remove them as we follow a deprecation cycle. we would need add these attributes and parses back and handle it in the config side with a handle deprecate function which would redirect all its functionality to the newer attributes. Could you also update the PR as to why it is being deprecated.

When an attribute needs to be deprecated, these are the steps to be followed:
- Mark that option deprecated, by including "This option has been deprecated by <enter new key name> and will be removed in a release after <enter date at least 2 years after current>." in the first line for it's description.
- Add documentation entry for new option. 
- Add facts and config support for it. Continue to render both old and new key in facts till removal date, but update example plays to ONLY use the new one.

You can refer this [PR](https://github.com/ansible-collections/cisco.ios/pull/687) for reference

Reply:
I have restored the param `mpls_vpn` in last PR (that's the only one I have removed).
During the loading of a config file, this parameter will be synced with the param `mpls.vpn.enable`